common/src/main/java/edu/snu/nemo/common/ir/vertex/SourceVertex.java

```diff
@@ -34,4 +34,9 @@
    * @throws Exception if fail to get.
    */
   public abstract List<Readable<O>> getReadables(int desiredNumOfSplits) throws Exception;
+
+  /**
+   * Clears internal states, must be called after getReadables().
```

Comment:
Could you add the following description in this comment too?
Since the meaning and context of 'internal states' of `SourceVertex` isn't clear by itself.

> Adds SourceVertex#clearInternalStates to clear the huge list of input splits held by BeamBoundedSourceVertex before sending the vertex to remote executors.